src/recorder/stream_recorder/stream_recorder.h

```diff
@@ -126,6 +129,11 @@ class StreamRecorder : public ForkAwareRecorder, private Noncopyable {
   std::atomic<int> pending_flush_counter_{0};
   int64_t num_spans_consumed_{0};
 
+  std::mutex shutdown_mutex_;
+  std::condition_variable shutdown_condition_variable_;
+  std::atomic<int> shutdown_counter_{0};
+  bool last_is_active_{true};
```

Comment:
I had a bit of trouble following what this variable does. Could you add a comment? Maybe add a comment on ShutdownWithTimeout to explain why it can fail even without a timeout.

Reply:
Added a comment